File: fibo.js
const fibo = n => {
  if (n <= 1) {
    return n;
  }
  else {
    return fibo(n-1) + fibo(n-2);
  }
}

const label = process.argv[2];
const n = process.argv[3];
let sum = 0;
for (let i = 0; i < 3; i++) {
  const start = Number(process.hrtime.bigint() / 100000n) / 10000;
  fibo(n);
  const end = Number(process.hrtime.bigint() / 100000n) / 10000;
  sum += end - start;
}
console.log(parseInt(sum / 3 * 10000) / 10000, label);

File: fibo.py
import sys
import time


def fibo(n):
    if n <= 1:
        return n
    else:
        return fibo(n-1) + fibo(n-2)


def main(threads, n):
    if threads == 1:
        fibo(n)
    else:
        from concurrent.futures import ThreadPoolExecutor

        with ThreadPoolExecutor(max_workers=threads) as executor:
            for _ in range(threads):
                executor.submit(fibo, n)

if __name__ == '__main__':
    label = sys.argv[1]
    threads = int(sys.argv[2])
    n = int(sys.argv[3])
    sum = 0
    for _ in range(3):
        start = time.time()
        main(threads, n)
        end = time.time()
        sum += end - start
    avg = sum / 3
    print(f'{avg:.04f} {label}')
